### PR TITLE
Fix #68281 [@types/react-datepicker] Add a new prop toggleCalendarOnIconClick

### PR DESCRIPTION
**Pull request by balajis-qb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://reactdatepicker.com/#example-toggle-calendar-open-status-on-click-of-the-calendar-icon)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/react-datepicker/index.d.ts [hunk 1]
```diff
@@ -198,6 +198,7 @@ export interface ReactDatePickerProps<
     excludeScrollbar?: boolean | undefined;
     enableTabLoop?: boolean | undefined;
     yearItemNumber?: number | undefined;
+    toggleCalendarOnIconClick?: boolean | undefined;
```
Comment (seckie) [inline types/react-datepicker/index.d.ts:201]: How about moving this line below the `todayButton?` line and making it alphabetical?

**Review by seckie — CHANGES_REQUESTED** [review]
toggleCalendarOnIconClick prop seems to need [v4.25.0](https://github.com/Hacker0x01/react-datepicker/releases/tag/v4.25.0) .
~~You need to update `version` field to `4.25.9999` in package.json.~~
-> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68437 will update v6.0. So you don't need to update it at this time.

FYI: https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library

File: types/react-datepicker/index.d.ts (hunk 1 above)
Comment (balajis-qb) [reply types/react-datepicker/index.d.ts]: Thank you for the suggestion.  I moved the `toggleCalendarOnIconClick` below the `todayButton` prop.  Kindly verify @seckie

**Review by seckie — APPROVED** [review]

**Comment by balajis-qb** [comment]
Ready to merge

**Pull request merged**